api/v1alpha3/condition_consts.go

```diff
@@ -47,6 +47,20 @@ const (
 	// while installing the container storage interface  addon; those kind of errors are usually transient
 	// the operation is automatically re-tried by the controller.
 	CSIProvisioningFailedReason = "CSIProvisioningFailed"
+
+	// FailureDomainsAvailableCondition documents the status of the failure domains
+	// associated to the VSphereCluster.
+	FailureDomainsAvailableCondition clusterv1.ConditionType = "FailureDomainsAvailable"
+
+	// FailureDomainsSkippedReason (Severity=Info) documents that some of the failure domain statuses
+	// associated to the VSphereCluster are reported as not ready.
+	FailureDomainsSkippedReason = "FailureDomainsSkipped"
+
+	// FailureDomainsNotReadyReason (Severity=Info) documents that some of the failure domains
+	// associated to the VSphereCluster are not reporting the Ready status.
+	// Instead of reporting a false ready status, these failure domains are still under the process of reconciling
+	// and hence not yet reporting their status.
+	FailureDomainsNotReadyReason = "FailureDomainsNotReady"
```

Comment:
`FailureDomainsNotReady` doesn't really convey that we're waiting on them to report status. It almost sounds like they reported a Ready: false. Maybe something like `WaitingForFailureDomainStatus`?